compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/intrinsics/ArrayConstructor.kt

```diff
@@ -34,10 +34,14 @@ object ArrayConstructor : IntrinsicMethod() {
     ): IrIntrinsicFunction {
         return object : IrIntrinsicFunction(expression, signature, context, expression.argTypes(context)) {
 
-            override fun invoke(v: InstructionAdapter, codegen: ExpressionCodegen, data: BlockInfo): StackValue =
-                codegen.generateCall(expression, this, data).let {
-                    return StackValue.onStack(Type.getObjectType("[" + AsmTypes.OBJECT_TYPE.internalName))
+            override fun invoke(v: InstructionAdapter, codegen: ExpressionCodegen, data: BlockInfo): StackValue {
+                // TODO fix this hack
```

Comment:
1) Why `with` is necessary here instead of just `codegen.generateCall(...).materialize()`?
2) I think `ExpressionCodegen.generateCall` should always materialize the generated value, at least until we have use cases to optimize away unneeded calls. Moreover, maybe we could have some sort of a convention that methods named `generateX` always return an already materialized value

Reply:
It used to be an extension method StackValue.materialize defined inside ExpressionCodegen. `with` is redundant now.